app/src/main/java/chat/rocket/android/wallet/create/ui/CreateWalletFragment.kt

```diff
@@ -0,0 +1,143 @@
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.disposables.Disposable
+import io.reactivex.rxkotlin.Observables
+import javax.inject.Inject
+import kotlinx.android.synthetic.main.fragment_create_wallet.*
+import timber.log.Timber
+
+class CreateWalletFragment:  Fragment(), CreateWalletView, android.support.v7.view.ActionMode.Callback {
+    @Inject lateinit var presenter: CreateWalletPresenter
+    private var actionMode: ActionMode? = null
+    private val disposables = CompositeDisposable()
+
+    companion object {
+        fun newInstance() = CreateWalletFragment()
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        AndroidSupportInjection.inject(this)
+        super.onCreate(savedInstanceState)
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? = container?.inflate(R.layout.fragment_create_wallet)
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        disposables.add(listenToChanges())
+    }
+
+    override fun onDestroyView() {
+        disposables.clear()
+        super.onDestroyView()
+    }
+
+    override fun onActionItemClicked(mode: ActionMode, menuItem: MenuItem): Boolean {
+
+        val walletName = editText_wallet_name.textContent
+        val passText = editText_password.textContent
+        val confirmPass = editText_confirm_password.textContent
+
+        return when {
+            walletName.isEmpty() -> {
+                showToast("Wallet must have a name.")
+                false
+            }
+            passText.isEmpty() -> {
+                showToast("Password cannot be empty.")
+                false
+            }
+            confirmPass != passText -> {
+                showToast("Passwords do not match.")
+                false
+            }
+            else -> {
+                mode.finish()
+                val walletName = editText_wallet_name.text.toString()
+                val password = editText_password.text.toString()
+                presenter.createNewWallet(walletName, password)
+                true
+            }
+        }
+    }
+
+    override fun onCreateActionMode(mode: ActionMode, menu: Menu?): Boolean {
+        mode.menuInflater.inflate(R.menu.password, menu)
+        mode.title = getString(R.string.action_confirm_create_wallet)
+        return true
+    }
+
+    override fun onPrepareActionMode(mode: ActionMode?, menu: Menu?): Boolean = false
+
+    override fun onDestroyActionMode(mode: ActionMode?) {
+        actionMode = null
+    }
+
+
+    override fun showWalletSuccessfullyCreatedMessage(mnemonic: String) {
+        showToast("Wallet was successfully created")
+        setMnemonic(mnemonic)
+    }
+
+    override fun showWalletCreationFailedMessage(error : String?) {
+        showToast("Failed to create wallet: " + error)
```

Comment:
In other areas of Rocket code, do they put strings like these toasts directly in the code, or do they have a string table somewhere so they can be translated to other languages?

Reply:
yes, looks like the toast message should go with the other string resources, I'll update this